### PR TITLE
bug-fix/internal-issue-#756

### PR DESCRIPTION
**Pull request by gnusi**

### CODE REVIEW

**Review by Dronplane — APPROVED** [review]
LGTM

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/iresearch_s-pr/154/

File: tests/search/tfidf_test.cpp [hunk 1]
```diff
@@ -101,6 +101,7 @@ TEST_P(tfidf_test, make_from_bool) {
     ASSERT_NE(nullptr, scorer);
     auto& tfidf = dynamic_cast<irs::tfidf_sort&>(*scorer);
     ASSERT_EQ(true, tfidf.normalize());
+    ASSERT_FALSE(tfidf.use_boost_as_score());
```
Comment (Dronplane) [inline tests/search/tfidf_test.cpp:104]:
```suggestion
    ASSERT_EQ(irs::tfidf_sort::BOOST_AS_SCORE(), tfidf.use_boost_as_score());
```

File: tests/search/tfidf_test.cpp [hunk 2]
```diff
@@ -1183,6 +1186,7 @@ TEST_P(tfidf_test, test_make) {
     ASSERT_NE(nullptr, scorer);
     auto& scr = dynamic_cast<irs::tfidf_sort&>(*scorer);
     ASSERT_FALSE(scr.normalize());
+    ASSERT_EQ(false, scr.use_boost_as_score());
```
Comment (Dronplane) [inline tests/search/tfidf_test.cpp:1189]:
```suggestion
    ASSERT_EQ(irs::tfidf_sort::BOOST_AS_SCORE(), tfidf.use_boost_as_score());
```

File: tests/search/tfidf_test.cpp [hunk 3]
```diff
@@ -1197,6 +1201,7 @@ TEST_P(tfidf_test, test_make) {
     ASSERT_NE(nullptr, scorer);
     auto& scr = dynamic_cast<irs::tfidf_sort&>(*scorer);
     ASSERT_EQ(true, scr.normalize());
+    ASSERT_EQ(false, scr.use_boost_as_score());
```
Comment (Dronplane) [inline tests/search/tfidf_test.cpp:1204]:
```suggestion
    ASSERT_EQ(irs::tfidf_sort::BOOST_AS_SCORE(), scr.use_boost_as_score());
```

File: tests/search/tfidf_test.cpp [hunk 4]
```diff
@@ -1211,6 +1216,7 @@ TEST_P(tfidf_test, test_make) {
...
```
Comment (Dronplane) [inline tests/search/tfidf_test.cpp:1219]:
```suggestion
    ASSERT_EQ(irs::tfidf_sort::BOOST_AS_SCORE(), scr.use_boost_as_score());
```

File: tests/search/tfidf_test.cpp [hunk 5]
```diff
@@ -1242,7 +1248,7 @@ TEST_P(tfidf_test, test_order) {
   *query.mutable_field() = "field";
 
   iresearch::order ord;
-  ord.add<iresearch::tfidf_sort>(true);
+  ord.add<iresearch::tfidf_sort>(true, false, true);
```
Comment (Dronplane) [inline tests/search/tfidf_test.cpp:1251]: ASSERT use_boost_as_score() is true here ? And maybe add test where setting boost_as_score = true affects results ?

**Pull request merged**